File: apps/web/src/api/coach.ts
import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query'
import { apiClient } from './client'

interface CoachReport {
  id: string
  session_id: string
  user_id: string
  diagnosis?: string
  interpretation?: string
  prescription?: string
  validation?: string
  next_best_move?: string
  linked_metrics?: Record<string, unknown>
  report_type: string
  language: string
  created_at: string
}

interface CoachReportCreate {
  session_id: string
  report_type?: string
  language?: string
}

interface ChatMessage {
  id: string
  user_id: string
  session_id?: string
  role: 'user' | 'assistant'
  content: string
  context?: Record<string, unknown>
  created_at: string
}

interface ChatRequest {
  content: string
  session_id?: string
  context?: Record<string, unknown>
}

export function useCoachReports(sessionId?: string) {
  const queryParams = sessionId ? `?session_id=${sessionId}` : ''
  
  return useQuery({
    queryKey: ['coach-reports', sessionId],
    queryFn: () =>
      apiClient.get<CoachReport[]>(`/coach/reports${queryParams}`),
  })
}

export function useCoachReport(id: string) {
  return useQuery({
    queryKey: ['coach-report', id],
    queryFn: () => apiClient.get<CoachReport>(`/coach/reports/${id}`),
    enabled: !!id,
  })
}

export function useGenerateReport() {
  const queryClient = useQueryClient()

  return useMutation({
    mutationFn: async (data: CoachReportCreate) => {
      return apiClient.post<CoachReport>('/coach/report', data)
    },
    onSuccess: (_, variables) => {
      queryClient.invalidateQueries({
        queryKey: ['coach-reports', variables.session_id],
      })
      queryClient.invalidateQueries({ queryKey: ['coach-reports'] })
    },
  })
}

export function useChatHistory(sessionId?: string) {
  const queryParams = sessionId ? `?session_id=${sessionId}` : ''
  
  return useQuery({
    queryKey: ['chat-history', sessionId],
    queryFn: () =>
      apiClient.get<ChatMessage[]>(`/coach/chat${queryParams}`),
  })
}

export function useSendChat() {
  const queryClient = useQueryClient()

  return useMutation({
    mutationFn: async (data: ChatRequest) => {
      return apiClient.post<ChatMessage>('/coach/chat', data)
    },
    onSuccess: (_, variables) => {
      queryClient.invalidateQueries({
        queryKey: ['chat-history', variables.session_id],
      })
      queryClient.invalidateQueries({ queryKey: ['chat-history'] })
    },
  })
}

File: apps/api/app/schemas/coach.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional, Any


class CoachReportCreate(BaseModel):
    session_id: UUID
    report_type: str = "session"
    language: str = "en"


class CoachReportResponse(BaseModel):
    id: UUID
    session_id: UUID
    user_id: UUID
    diagnosis: Optional[str]
    interpretation: Optional[str]
    prescription: Optional[str]
    validation: Optional[str]
    next_best_move: Optional[str]
    linked_metrics: Optional[dict[str, Any]]
    report_type: str
    language: str
    created_at: datetime

    class Config:
        from_attributes = True


class ChatMessageCreate(BaseModel):
    content: str
    session_id: Optional[UUID] = None
    context: Optional[dict[str, Any]] = None


class ChatMessageResponse(BaseModel):
    id: UUID
    user_id: UUID
    session_id: Optional[UUID]
    role: str
    content: str
    context: Optional[dict[str, Any]]
    created_at: datetime

    class Config:
        from_attributes = True
